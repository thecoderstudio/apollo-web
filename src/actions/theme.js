function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
