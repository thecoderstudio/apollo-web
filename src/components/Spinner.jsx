import React from 'react';
import styled, { withTheme } from 'styled-components';
import Loader from 'react-loader-spinner';
import PropTypes from 'prop-types';

const propTypes = {
  type: PropTypes.string
};

const defaultProps = {
  type: 'TailSpin'
};

const StyledLoader = styled(Loader)`
  position: fixed;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  height: 100px;
  width: 100px;
`;

function Spinner(props) {
  return (
    <StyledLoader
      type={props.type}
      color={props.theme.primary}
    />
  );
}

Spinner.propTypes = propTypes;
Spinner.defaultProps = defaultProps;

export default withTheme(Spinner);
