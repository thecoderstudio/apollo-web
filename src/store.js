import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import storage from 'redux-persist/lib/storage';
import mapTransformer from './transformers/map';
import rootReducer from './reducers/root';

const persistConfig = {
  key: 'root',
  whitelist: ['auth', 'agent'],
  transforms: [mapTransformer({ whitelist: 'agent' })],
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
