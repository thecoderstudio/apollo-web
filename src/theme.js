export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
