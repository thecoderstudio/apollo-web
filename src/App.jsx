import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import ProtectedRoute from './components/ProtectedRoute';
import NavBar from './components/NavBar';
import Notifications from './components/notifications/Notifications';
import Dashboard from './pages/Dashboard';
import Welcome from './pages/Welcome';
import AgentDetail from './pages/AgentDetail';
import TerminalPage from './pages/TerminalPage';
import Admin from './pages/Admin';
import NotFound from './pages/NotFound';
import { darkTheme } from './theme';

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=B612:wght@400;700&family=Roboto:wght@300;400&display=swap');

  html, body {
    margin: 0px;
    padding: 0px;
    height: 100%;
    width: 100%;
    font-weight: 500;
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.black};
  }

  body {
    font-family: 'Roboto', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'B612', sans-serif;
  }
`;

const Content = styled.div`
  height: 100%;
  width: 100%;
`;

function App(props) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Content>
            {props.authenticated && props.hasChangedPassword && <NavBar />}
            <Switch>
              <ProtectedRoute exact path="/" component={Dashboard} fallbackComponent={Welcome} />
              <ProtectedRoute exact path="/agent/:agentId/shell" component={TerminalPage} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/agent/:agentId" component={AgentDetail} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/admin" component={Admin} fallbackComponent={NotFound} role="admin" />
              <Route component={NotFound} />
            </Switch>
            <Notifications />
            <GlobalStyle />
          </Content>
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
}

export default connect(
  (state) => ({
    authenticated: state.authenticated,
    hasChangedPassword: state.currentUser.hasChangedInitialPassword
  })
)(App);
