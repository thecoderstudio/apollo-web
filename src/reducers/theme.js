import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
