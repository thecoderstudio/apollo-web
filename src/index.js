import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
