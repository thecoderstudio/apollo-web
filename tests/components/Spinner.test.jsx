import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Spinner from '../../src/components/Spinner';
import { darkTheme } from '../../src/theme';

test('Spinner renders correctly', () => {
  const tree = renderer
    .create(<ThemeProvider theme={darkTheme}><Spinner /></ThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
