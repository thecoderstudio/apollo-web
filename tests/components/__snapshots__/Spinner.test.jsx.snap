// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
.c0 {
  position: fixed;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 100px;
  width: 100px;
}

<div
  aria-busy="true"
  className="c0"
  style={Object {}}
>
  <svg
    aria-label="audio-loading"
    height={80}
    viewBox="0 0 38 38"
    width={80}
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <linearGradient
        id="a"
        x1="8.042%"
        x2="65.682%"
        y1="0%"
        y2="23.865%"
      >
        <stop
          offset="0%"
          stopColor="#02A3EE"
          stopOpacity="0"
        />
        <stop
          offset="63.146%"
          stopColor="#02A3EE"
          stopOpacity=".631"
        />
        <stop
          offset="100%"
          stopColor="#02A3EE"
        />
      </linearGradient>
    </defs>
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        transform="translate(1 1)"
      >
        <path
          d="M36 18c0-9.94-8.06-18-18-18"
          id="Oval-2"
          stroke="#02A3EE"
          strokeWidth="2"
        >
          <animateTransform
            attributeName="transform"
            dur="0.9s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </path>
        <circle
          cx="36"
          cy="18"
          fill="#fff"
          r={1}
        >
          <animateTransform
            attributeName="transform"
            dur="0.9s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </circle>
      </g>
    </g>
  </svg>
</div>
`;
